dataset.py

```diff
@@ -26,34 +27,71 @@ def get_next_n_queries(self, n):
         return [self.dataset_list[i] for i in next_n_indices]
 
 
-def read_jsonlines(
-    filename, max_queries=1000, max_input_tokens=1024, max_output_tokens=1024
+def initialize_dataset(
+    model_name, filename,
+    max_queries=1000, 
+    min_input_tokens=0, 
+    max_input_tokens=1024, 
+    max_output_tokens=1024, 
+    max_sequence_tokens=4096
 ):
+    
+    prompt_format = get_format_string(model_name)
     with open(filename, "r", encoding="utf-8") as file:
         total_queries = 0
-        for idx, line in enumerate(file):
-            if idx == 0:
-                # skip the first line, it contains metadata
-                continue
+
+        # [1:] to skip the first line, it contains metadata
+        lines = file.readlines()[1:]
+        random.Random(dataset_seed).shuffle(lines)
+
+        for line in lines:
             # Load each line as a JSON object
             try:
                 json_object = json.loads(line.strip())
             except json.JSONDecodeError as e:
                 logging.error("Error decoding JSON in file %s %s", filename, e)
                 continue
             try:
-                input_tokens = json_object["tok_input_length"]
-                input_data = json_object["input"]
-                output_tokens = input_data["min_new_tokens"]
+                input_tokens = int(json_object["tok_input_length"])
+                output_tokens = int(json_object["tok_output_length"])
```

Comment:
you don't really need to `int()` cast here, json gives it to you in the right format
(except if some datasets are inconsistent ...)